src/view_pos_transform.hpp

```diff
@@ -0,0 +1,184 @@
+// ==========================================================================
+//                                  lambda
+// ==========================================================================
+// Copyright (c) 2013-2019, Hannes Hauswedell <h2 @ fsfe.org>
+// Copyright (c) 2016-2020, Knut Reinert and Freie Universität Berlin
+// All rights reserved.
+//
+// This file is part of Lambda.
+//
+// Lambda is Free Software: you can redistribute it and/or modify it
+// under the terms found in the LICENSE[.md|.rst] file distributed
+// together with this file.
+//
+// Lambda is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//
+// ==========================================================================
+// view_pos_transform.hpp
+// ==========================================================================
+
+#pragma once
+
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/range/views/detail.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/std/ranges>
+
+// The return type of views::pos_transform
+template <std::ranges::view urng_t, typename pos_transform_t, typename size_transform_t>
+class view_pos_transform : public ranges::view_base
+{
+
+private:
+    // The data members of view_pos_transform.
+    urng_t                              urange;
+    std::shared_ptr<pos_transform_t>    pos_transform = nullptr;
+    std::shared_ptr<size_transform_t>   size_transform = nullptr;
```

Comment:
Can you replace `std::shared_ptr` with `seqan3::semiregular_box_t`? That might make the performance better.